src/tools/generate_workspace/src/main/java/com/google/devtools/build/workspace/GenerateWorkspace.java

```diff
@@ -59,7 +60,13 @@ public static void main(String[] args) {
       return;
     }
 
-    GenerateWorkspace workspaceFileGenerator = new GenerateWorkspace(options.outputDir);
+    if ((options.buildOutputFile.isEmpty() && options.workspaceOutputFile.isEmpty())
+        != options.outputDir.isEmpty()) {
```

Comment:
Wait, do you want people to be able to specify --output_dir and one of --build_file or --workspace_file?